Address points raised during IESG review and Directorate Reviews

paulwouters opened the pull request.

paulwouters committed:
Update for IESG ballot comments by Paul Wouters

See https://mailarchive.ietf.org/arch/msg/dnsop/tVlyDhQvie03hq8TwU4ib2M8AZg/

paulwouters committed:
Update for IESG ballot comments by Éric Vyncke

See: https://mailarchive.ietf.org/arch/msg/dnsop/oDc5KX_FtY3g8qbHC4dQk1o3FNc/

paulwouters committed:
Update for IESG ballot comments by Lars Eggert

See: https://mailarchive.ietf.org/arch/msg/dnsop/BwQNPQ88u4vo-UoU2tUMzvN-IWI/

paulwouters committed:
Update for IESG ballot DISCUSSes by Martin Duke

See: https://mailarchive.ietf.org/arch/msg/dnsop/sgoSHMc_boOWxpIcx1YfnoDS5xk/

paulwouters committed:
Updates for Barry Leiba's ARTART review

See: https://datatracker.ietf.org/doc/review-ietf-dnsop-avoid-fragmentation-15-artart-lc-leiba-2023-10-23/

evyncke commented on draft-ietf-dnsop-avoid-fragmentation.mkd:
```diff
@@ -224,14 +239,15 @@ R1. UDP responders SHOULD send DNS responses
   the interface MTU,
-  and the RECOMMENDED maximum DNS/UDP payload size 1400.
+  and the RECOMMENDED maximum DNS/UDP payload size 1400 as a safe value based
+  on various meassurement experiments such as [DefragmentingDNS] and [Huston2021].
```
Thanks

evyncke commented on draft-ietf-dnsop-avoid-fragmentation.mkd:
```diff
@@ -429,6 +447,10 @@ default path MTU size and requestor's maximum UDP payload size.
 
+- [DefragmentingDNS] concluded there is no single "magical" MTU message
+  size for DNS resolver implementations, but pointed out some safe values
+  for stub resolvers at 1400 for IPv4 and 1280 for IPv6.
```
Above it says "DNS/UDP payload is 1400" and here is says "MTU 1400" (if not mistaken --getting late here-- MTU is for IP header and all the rest, i.e., a difference of 20 octets

paulwouters replied:
That is correct. Some references talk about the EDNS0 buffer size and some talk about the final MTU size. I did not want to modify the references with an interpretation.

kfujiwara commented:
Sorry, I could not update this draft soon. I could not merge my commits. Self merge is not good.
R2: Robert Wilton's DISCUSS proposed another fix. I prefer this. Can you change and re-commit ? Then, I can merge.

    R2. Where supported, UDP responders SHOULD set
    IP "Don't Fragment flag (DF) bit" [RFC0791] on IPv4.

paulwouters committed:
followup for Éric Vyncke's comment on IPv6 fragmentation.

Also fixup a MAY -> SHOULD as per Kazunori Fujiwara's comment.

paulwouters committed:
Update for Lars Eggert IESG comments

Resolves: https://github.com/ietf-wg-dnsop/draft-ietf-dnsop-avoid-fragmentation/issues/23

paulwouters committed:
Update for Lars Eggert comment 5

Resolves: https://github.com/ietf-wg-dnsop/draft-ietf-dnsop-avoid-fragmentation/issues/21

paulwouters committed:
Update for Christer Holmberg GEN review comment

Resolves: https://github.com/ietf-wg-dnsop/draft-ietf-dnsop-avoid-fragmentation/issues/16

paulwouters committed:
Update for Orie Steele / Joe Abley "zone operator"

Use "DNS content operator".

Resvoles: https://github.com/ietf-wg-dnsop/draft-ietf-dnsop-avoid-fragmentation/issues/15
(at least the non-meta question part of it)

paulwouters committed:
Update for Martin Duke timeout issue

Resolves: https://github.com/ietf-wg-dnsop/draft-ietf-dnsop-avoid-fragmentation/issues/10

paulwouters committed:
Update for Robert Wilton's DISCUSS on section 3.2

Resolves: https://github.com/ietf-wg-dnsop/draft-ietf-dnsop-avoid-fragmentation/issues/8

paulwouters committed:
Update for Robert Wilton's DISCUSS 2

Resolves: https://github.com/ietf-wg-dnsop/draft-ietf-dnsop-avoid-fragmentation/issues/6

paulwouters committed:
Update for Donald Eastlake SECDIR review

Resolves: https://github.com/ietf-wg-dnsop/draft-ietf-dnsop-avoid-fragmentation/issues/4

paulwouters committed:
Update for Donald Eastlake's SECDIR implementation status note

paulwouters committed:
Update for Donald Eastlake SecDir Security comments

Resolves: https://github.com/ietf-wg-dnsop/draft-ietf-dnsop-avoid-fragmentation/issues/2

This does not pick up the DNS transaction suggestion, as those
transports are not for the public resolver resolution path.

vixie commented:
it would have been easier to review these changes as topic-specific pull requests. that's why we have separate issues.

it now appears that some of the changes from the wouters PR are also present on the fujiwara PR. if that's the case is there any need to also review the wouters PR?

on-list, the discussion about operating system support for the DF bit and whether we ought say SHOULD rather than MUST since it is not universally possible to set this bit, ended with MUST since it's a goal and we can expect gradual trending towards compliance. it is difficult to determine from these two pull requests whether that outcome has been preserved.

i also entered one comment on the fujiwara PR, concerning the removal of the path MTU discovery/override section.

moonshiner commented:
My only concern here from the IESG feedback is some of these MAY were based on working group and implementer feedback, and changing them to SHOULD may encounter pushback from the working group, or worse ,the implementers finding this to be non useful.  I'll need to go back over my notes
